### PR TITLE
Add alpha 3 changelog.

### PR DESCRIPTION
**Pull request by vpetrovykh**

### CODE REVIEW

File: docs/changelog/1_0_a3.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+
+Rust
+====
+
+We've started the process of rewriting some of the
+performance-critical EdgeDB internals to Rust. So far we have
+converted the lexer as well as the GraphQL parser.
+
+Additionally, we're also making EdgeDB command-line tools using Rust.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:16]: making...using --- this sentence needs a rewrite. I'd also mention the command line tools first and then talk about graphql.

File: docs/changelog/1_0_a3.rst [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+Notably the REPL has been converted and now features more commands:
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:17]: All tools have been converted. The goal of the conversion is ease of distribution. The tools are a single binary now and can run on multiple different platforms, including Windows.

File: docs/changelog/1_0_a3.rst [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+.. code-block::
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:19]: I don't think we need to include this print-out.  Besides, there are only a few new commands (dump/restore?)

Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:19]: History, edit, set, recent error are also new. Plus the commands themselves changed (and we've had them listed last time).

File: docs/changelog/1_0_a3.rst [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+    Connection
+      \c, \connect [DBNAME]    Connect to database DBNAME
+
+    Development
+      \E                       show most recent error message at
+                               maximum verbosity
+      \pgaddr                  show the network addr of the postgres
+                               server
+      \psql                    open psql to the current postgres
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:58]: This is also old output, we dropped this command from help text.

File: docs/changelog/1_0_a3.rst (hunk 1 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:16]: OK, I'll swap order.

File: docs/changelog/1_0_a3.rst [hunk 5]
```diff
@@ -0,0 +1,158 @@
+===========
+1.0 Alpha 3
+===========
+
+This changelog summarizes new features and breaking changes in
+EdgeDB 1.0 alpha 3.
+
+
+Rust
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:9]: I'd call this section CLI

File: docs/changelog/1_0_a3.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:17]: I'll add the comment about distribution, etc.

File: docs/changelog/1_0_a3.rst [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+  same Postgres cluster.
+* Protocol: implement ``JSON_ELEMENTS`` IO format for responses
+  (:eql:gh:`#1169`).
+
+
+Misc
+====
+
+* Add cheatsheet.
+* Add an interactive tutorial.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:154]: Not related to EdgeDB changelog.

File: docs/changelog/1_0_a3.rst [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+* Make it possible to use different EdgeDB servers connected to the
+  same Postgres cluster.
+* Protocol: implement ``JSON_ELEMENTS`` IO format for responses
+  (:eql:gh:`#1169`).
+
+
+Misc
+====
+
+* Add cheatsheet.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:153]: ... to the documentation.

File: docs/changelog/1_0_a3.rst [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+* Add an interactive tutorial.
+* Strictly type-annotate the core schema module.
+* Clean-up and improve schema and introspection code.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:156]: Please reference Elvis' PR.

File: docs/changelog/1_0_a3.rst [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+* A number of updates and bugfixes for Python and JavaScript drivers,
+  including the support of ``edgedb-js`` for browsers without BigInt.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:158]: I wouldn't mention browser support in `edgedb-js` -- it's not even alpha stage.

Comment (1st1) [reply docs/changelog/1_0_a3.rst:158]: Also if there are some concrete issues/prs for drivers please reference them.

File: docs/changelog/1_0_a3.rst (hunk 5 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:9]: I thought that "Rust" might highlight the nature of the change better, but we can make it CLI instead (and move the comment about internals rewritten in Rust to "Server")

File: docs/changelog/1_0_a3.rst (hunk 4 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:58]: Odd, I used my REPL help to get this (which should be pretty fresh). I'll pull the latest version.

File: docs/changelog/1_0_a3.rst (hunk 3 above)
Comment (1st1) [reply docs/changelog/1_0_a3.rst:19]: I'd just enumerate the new commands then, briefly. Like "there are new commands to manage and navigate history: `\blah1`, `\blah2`"

File: docs/changelog/1_0_a3.rst (hunk 5 above)
Comment (1st1) [reply docs/changelog/1_0_a3.rst:9]: Well, we want the changelog to be succinct and to the point. The section is about CLI, and its rewrite to Rust is an important point, but not the only point.

**Comment by vpetrovykh** [comment]
> I don't see that we now extract constants from EdgeQL and GraphQL queries mentioned anywhere.

I suppose that constant extraction can go into "Sever" section rather than EdgeQL/GraphQL since fundamentally it's not so much about the languages as it is about more efficient caching for the server.

File: docs/changelog/1_0_a3.rst (hunk 8 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:156]: There's more than one PR that cleans up schema code (like making collections more well-behaved, or generalizing introspection). Collectively they make schema and introspection work better, but I'm not sure that pointing to any individual PR is really all that enlightening.

Comment (1st1) [reply docs/changelog/1_0_a3.rst:156]: Please reference this pr https://github.com/edgedb/edgedb/pull/1408

> I'm not sure that pointing to any individual PR is really all that enlightening.

I'm not then sure why we are referencing any commits/prs?

File: docs/changelog/1_0_a3.rst [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+===========
+
+This changelog summarizes new features and breaking changes in
+EdgeDB 1.0 alpha 3.
+
+
+CLI
+===
+
+We've converted EdgeDB command-line tools to using Rust. The tools are
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:12]:
```suggestion
The EdgeDB command-line tools are now written in Rust. The tools are
```

File: docs/changelog/1_0_a3.rst (hunk 2 above)
Comment (1st1) [reply docs/changelog/1_0_a3.rst:16]: "Notably the REPL has been converted and now features more commands:" -- this sentence is not needed anymore?

Victor, please do a complete read of this file and fix it.

File: docs/changelog/1_0_a3.rst [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+====
+
+* Add cheatsheet to documentation.
+* Strictly type-annotate the core schema module.
+* Clean-up and improve schema and introspection code to make the
+  internal APIs more consistent (:eql:gh:`#1408`).
+* Update the `edgedb-js <https://github.com/edgedb/edgedb-js>`_ driver
+  to v0.7.3.
+* Update the `edgedb-python <https://github.com/edgedb/edgedb-python>`_
+  driver to v0.8.0a1.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:137]: 0.8.0, without a1.  We'll release a stable version today/tomorrow.

File: docs/changelog/1_0_a3.rst [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+  same Postgres cluster (:eql:gh:`#1197`).
+* Protocol: implement ``JSON_ELEMENTS`` IO format for responses
+  (:eql:gh:`#1169`).
+
+
+
+Misc
+====
+
+* Add cheatsheet to documentation.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:130]: I'd hyperlink the "cheatsheet"

File: docs/changelog/1_0_a3.rst [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+* Change the auto-generated type name template to use "\_Type" suffix
+  (:eql:gh:`#1175`).
+* Fix incorrect reflection of enums (:eql:gh:`#1227`).
+* Fix deeply nested insert issue (:eql:gh:`#1243`).
+
+
+Standard Library
+================
+
+* Add ``sys::get_current_database()``
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:106]: Can you use `eql:func:` here?

File: docs/changelog/1_0_a3.rst [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+  (:eql:gh:`#1352`, :eql:gh:`#1355`)
+* Allow tuple arrays in schema definitions (:eql:gh:`73125882`).
+* Get rid of ``schema::CompoundType`` in favor of ``is_compound_type``
+  property (:eql:gh:`c30c1c4f`).
+* Add :eql:type:`anytype` and :eql:type:`anytuple` as proper abstract
+  types to the schema (:eql:gh:`affb65de`).
+* ``schema::Constraint`` no longer has ``args``, use ``params`` instead
+  (:eql:gh:`14cec6f1`).
+* Make ``std::Object`` the implicit base type for all user-defined
+  object types only.
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:72]:
```suggestion
  object types.
```

File: docs/changelog/1_0_a3.rst [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+including Windows.
+
+Notably the REPL has been converted and now features more commands:
+
+.. code-block::
+
+      \dump FILENAME           dump current database into a file
+      \restore FILENAME        restore the database from file into
+                               the current one
+      \create-database DBNAME  create a new database
```
Comment (1st1) [inline docs/changelog/1_0_a3.rst:23]: Please drop `\create-database`

**Comment by haikyuu** [comment]
Will Migrations be part of alpha3 ?

**Comment by elprans** [comment]
> Will Migrations be part of alpha3 ?

Alas, the implementation isn't ready yet.  We decided to cut a release and not wait until RFC1000 is fully implemented to give other features and fixes the opportunity to get tested.  That said, we'll be switching to time-based 6-week release cadence for the alphas, betas and rcs from now on.

File: docs/changelog/1_0_a3.rst (hunk 2 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst]: Reworded.

File: docs/changelog/1_0_a3.rst (hunk 10 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst]: OK

File: docs/changelog/1_0_a3.rst (hunk 11 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst]: OK

File: docs/changelog/1_0_a3.rst (hunk 12 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst]: OK, done.

File: docs/changelog/1_0_a3.rst (hunk 13 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst]: Done

File: docs/changelog/1_0_a3.rst (hunk 14 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst:71]: I put "only" there to emphasize how this is different from the `BaseObject` that's listed onthe next line.

File: docs/changelog/1_0_a3.rst (hunk 15 above)
Comment (vpetrovykh) [reply docs/changelog/1_0_a3.rst]: Done

**Comment by 1st1** [comment]
Thanks Victor!

**Pull request merged**